### PR TITLE
BE-22-34-35 create internship/internship type/subject entity

### PR DESCRIPTION
**Pull request by nazfurdychka**

### CODE REVIEW

File: src/main/java/com/exadel/project/internship/entity/Internship.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    private List<Subject> subjects = new ArrayList<>();
+
+    @Column(name = "start_date", nullable = false)
+    private Date start_date;
+
+
+    @Column(name = "end_date", nullable = false)
+    private Date end_date;
+
+
```
Comment (serkushner) [inline src/main/java/com/exadel/project/internship/entity/Internship.java:39]: Minor: pls remove redundant empty space

File: src/main/java/com/exadel/project/internship/entity/Internship.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    @Enumerated(EnumType.STRING)
+    @Column(name = "format")
+    private Format format;
+
+    @ManyToOne
+    @JoinColumn(name = "type_id")
+    private InternshipType internship_type;
+
+
```
Comment (serkushner) [inline src/main/java/com/exadel/project/internship/entity/Internship.java:65]: The same

File: src/main/java/com/exadel/project/internship_type/entity/InternshipType.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Entity
+@Table(name = "internship_type")
+@Setter
+@Getter
```
Comment (serkushner) [inline src/main/java/com/exadel/project/internship_type/entity/InternshipType.java:16]: Pls replace with lombok `@Data` annotation

File: src/main/java/com/exadel/project/internship/entity/InternshipType.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import javax.validation.constraints.NotBlank;
+import java.util.ArrayList;
+import java.util.List;
+
+@Data
+@Entity
+@Table(name = "internship_type")
+public class InternshipType {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
```
Comment (serkushner) [inline src/main/java/com/exadel/project/internship/entity/InternshipType.java:16]: Maybe `GenerationType.IDENTITY`?

**Review by serkushner — CHANGES_REQUESTED** [review]

File: src/main/java/com/exadel/project/internship/entity/Internship.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    @ManyToMany
+    @JoinTable(name = "internship_subject",
+            joinColumns = @JoinColumn(name = "internship_id"),
+            inverseJoinColumns = @JoinColumn(name = "subject_id")
+    )
+    private List<Subject> subjects = new ArrayList<>();
+
+    @Column(name = "start_date", nullable = false)
+    private Date start_date;
+
```
Comment (AlKaliada) [inline src/main/java/com/exadel/project/internship/entity/Internship.java:34]: pls rename in camelCase

File: src/main/java/com/exadel/project/internship/entity/Internship.java (hunk 1 above)
Comment (nazfurdychka) [reply src/main/java/com/exadel/project/internship/entity/Internship.java:37]: done

File: src/main/java/com/exadel/project/internship/entity/Internship.java (hunk 2 above)
Comment (nazfurdychka) [reply src/main/java/com/exadel/project/internship/entity/Internship.java:63]: done

File: src/main/java/com/exadel/project/internship_type/entity/InternshipType.java (hunk 3 above)
Comment (nazfurdychka) [reply src/main/java/com/exadel/project/internship_type/entity/InternshipType.java:16]: done

File: src/main/java/com/exadel/project/internship/entity/Internship.java (hunk 5 above)
Comment (nazfurdychka) [reply src/main/java/com/exadel/project/internship/entity/Internship.java:32]: done

File: src/main/java/com/exadel/project/internship/entity/Internship.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+    @Column(name = "end_date", nullable = false)
+    private Date end_date;
```
[the remark this answers is not in the document]
Comment (nazfurdychka) [reply src/main/java/com/exadel/project/internship/entity/Internship.java]: done

**Review by serkushner — APPROVED** [review]

**Review by AlKaliada — APPROVED** [review]

**Pull request merged**